### ISSUE TITLE
How to specify the automl sdk version to be used?

### ISSUE THREAD

**Issue by robert4os** [Azure/MachineLearningNotebooks#842]
Is there a way to specify the automl version (i.e. via Conda dependencies)?

I am getting different results (even errors) because of different automl versions beeing used depending on the compute target (e.g. Training Cluster and local Compute Instances).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0bc2b21e-6b1a-cb94-2857-147177a29d7c
* Version Independent ID: d14620a6-a2f6-49f1-632e-73903d41de8c
* Content: [azureml.train.automl.automlconfig.AutoMLConfig class - Azure Machine Learning Python](https://docs.microsoft.com/en-us/python/api/azureml-train-automl-client/azureml.train.automl.automlconfig.automlconfig?view=azure-ml-py)
* Content Source: [AzureML-Docset/stable/docs-ref-autogen/azureml-train-automl-client/azureml.train.automl.automlconfig.AutoMLConfig.yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr/blob/live/AzureML-Docset/stable/docs-ref-autogen/azureml-train-automl-client/azureml.train.automl.automlconfig.AutoMLConfig.yml)
* Service: **machine-learning**
* Sub-service: **core**
* GitHub Login: @j-martens
* Microsoft Alias: **jmartens**

**Comment by ram-msft** [comment]
@robert4os Thanks for the question. We are investigating the issue and will update you shortly.

**Comment by ram-msft** [comment]
@robert4os Can you please add more details about the errors that you are getting.
Please follow the below to specify the automl version.
aml_conda_packages = ['pandas','scikit-learn','pyarrow','pytz','psutil']
aml_pip_packages = ['azureml-sdk', 'azureml-dataprep', 'azureml-train-automl==1.0.85']
aml_run_config.environment.python.conda_dependencies = CondaDependencies.create(
    conda_packages = aml_conda_packages, 
    pip_packages = aml_pip_packages, 
    pin_sdk_version=False)
registration_input = model_data

model_registration_step = PythonScriptStep(name='Register Model',
                                 script_name='register.py',
                                 compute_target=aml_compute,
                                 source_directory=scripts_folder,
                                 arguments=['--model_name', aml_trained_model_name, 
                                            '--model_path', model_data,
                                            '--model_path_folder', aml_trained_model_folder,
                                            '--model_date_path', string_pipeline_param,
                                            '--runconfig',aml_run_config,
                                            '--trained_model_name', aml_trained_model_name
                                           ],
                                 inputs=[model_data],
                                 allow_reuse=False

**Comment by robert4os** [comment]
Hi Ram,

thank you for your response.

I am using currently using

automl_config = AutoMLConfig(training_data=train_data, **automl_settings)
experiment = Experiment(ws, experiment_name)
 run = experiment.submit(automl_config, show_output=verbose)

and there I did not find an option to specify the azurml[ml]-sdk to use on
the target.

Are you suggesting to wrap the call above in a Script and execute via
PythonScriptStep ?
Is there a doc or notebook to demonstrate this? Especially using it to
execute Automl?

The errors I found out, where unrelated to the version. Still it woul be
very helpful to have control over the SDK version even for Automl calls
like above.

Thank you and best regards,
Robert
